src/LanguageServer/Protocol/Extensions/Extensions.cs

```diff
@@ -175,12 +175,29 @@ public static T FindDocumentInProjectContext<T>(this ImmutableArray<T> documents
             // This branch should be removed when we're using the LSP based navbar in all scenarios.
 
             var solution = documents.First().Project.Solution;
+
             // Lookup which of the linked documents is currently active in the workspace.
             var documentIdInCurrentContext = solution.Workspace.GetDocumentIdInCurrentContext(documents.First().Id);
-            return documentGetter(solution, documentIdInCurrentContext);
+
+            var document = documentGetter(solution, documentIdInCurrentContext);
+
+            // When a misc document is loaded into the host workspace, it's project file path is the same as the document file path.
+            // We always want to prefer the non-misc document.
+            if (IsMiscDocumentInHostWorkspace(document))
+            {
+                var nonMisc = documents.First(d => d.Id != document.Id);
+                return nonMisc;
+            }
+
+            return document;
         });
     }
 
+    public static bool IsMiscDocumentInHostWorkspace(this TextDocument document)
+    {
+        return document.Project.Solution.WorkspaceKind == WorkspaceKind.Host && document.Project.FilePath != null && document.Project.FilePath == document.FilePath;
```

Comment:
This was the best I could come up with to determine if we had a FBP without needing access to the filebased program project system.

Seems reasonable to me to have some indicator on the FBP `Project` indicating it may not be a normal project

Reply:
There is a `/feature:FileBasedProgram`, but nothing stops user from placing that on an ordinary project.

Reply:
Yeah ideally it would be something we control when we create the project (or document).  Tags sound nice, but I don't think we have anything like that.